prereise/gather/griddata/hifld/data_process/generators.py

```diff
@@ -84,25 +86,198 @@ def map_generator_to_bus_by_sub(generator, bus_groupby):
         return bus_groupby.get_group(generator.sub_id)["baseKV"].idxmin()
 
 
-def build_plant(bus, substations):
+def estimate_heat_rate_curve(
+    generator, epa_ampd_groupby, crosswalk_translation, min_unique_x=3, min_points=24
+):
+    """Estimate a generator's heat rate curve if data are present in the EPA AMPD data,
+    otherwise return NA.
+
+    :param pandas.Series generator: one generating unit from data frame.
+    :param pandas.GroupBy epa_ampd_groupby: data frame of EPA AMPD samples, grouped by
+        plant ID and unit ID.
+    :param pandas.Series crosswalk_translation: mapping of EIA keys to EPA keys.
+    :param int min_unique_x: minimum number of unique x points needed to fit a curve to.
+    :param int min_points: minimum number points needed to fit a curve to.
+    :return: (*tuple*) -- if possible, quadratic coefficients:
+        c0 (constant), c1 (linear), and c2 (quadratic); respectively. Otherwise, return
```

Comment:
Not sure whether we should replace c0, c1, c2 to h0, h1, h2, given the latter is the `return_index`?

Reply:
You're right. Initially I had the heat rates being fit as `c0, c1, c2`, and then I changed the heat rates to `h` and the cost curves to `c`. Fixed.